### PR TITLE
Added Make Archive + URL Encode Actions

### PR DESCRIPTION
**Pull request by gcordalis**
**Checks**

- [x] I've checked there are no linting errors.
- [x] I've checked the code is still at 100% test coverage.

**Added Actions (if relevant)**

- Make Archive
- URL Encode

**Are you happy to be listed as a contributor on [Shortcuts.fun](https://shortcuts.fun)?**

Yes

**Any other information / comments**

Fingers crossed this is all good <3

### CODE REVIEW

File: src/actions/makeArchive.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *   format: '.zip',
+ *   name: 'zipName',
+ * });
+ * ```
+ */
+const makeArchive = (
+  options: {
+    /** The operation to perform */
+    format?: WFArchiveFormat,
+    name?: WFSerialization | string,
```
Comment (xAlien95) [inline src/actions/makeArchive.ts]: Great pull request! Could you add a comment for the name parameter too?

Comment (xAlien95) [reply src/actions/makeArchive.ts]: @gcordalis, the Travis error is about a trailing whitespace after the comment that you added. Make sure there aren’t spaces after lines and check whether that line has some spaces at the end.

As a side note: Atom automatically removes trailing whitespaces on save 🙂

Comment (gcordalis) [reply src/actions/makeArchive.ts]: Yeah, I use VS Code when on desktop/laptop but I do a bit of work on an iPad. I haven't got a good workflow for this on here yet. Need to get that sorted :)

File: src/actions/makeArchive.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { withActionOutput } from '../utils';
+
+import WFArchiveFormat from '../interfaces/WF/WFArchiveFormat';
+import WFSerialization from '../interfaces/WF/WFSerialization';
+import WFWorkflowAction from '../interfaces/WF/WFWorkflowAction';
+
+/**
+ * Make Archive Action. Makes an archive out of the files passed as input.
```
Comment (joshfarrant) [inline src/actions/makeArchive.ts]: Could you add the rest of the description to this comment please? It should read:

```
 * Make Archive Action. Makes an archive out of the files passed as input. Supports creating zip, tar.gz, tar.bz2, tar, gzip, cpio or iso archives.
```

Thanks!

File: src/actions/urlEncode.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { withActionOutput } from '../utils';
+
+import WFEncodeMode from '../interfaces/WF/WFEncodeMode';
+import WFWorkflowAction from '../interfaces/WF/WFWorkflowAction';
+
+/**
+ * URL Encode Action. Encodes or decodes text passed into the action to be suitable for inclusion in
+ * a URL by adding. or removing percent escapes when appropriate
```
Comment (joshfarrant) [inline src/actions/urlEncode.ts]: Rogue '.' after 'adding', and also needs a '.' after 'appropriate' 🙂

File: src/interfaces/WF/WFArchiveFormat.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+type WFArchiveType = (
```
Comment (joshfarrant) [inline src/interfaces/WF/WFArchiveFormat.ts]: Should the variable name be `WFArchiveFormat`?

Comment (gcordalis) [reply src/interfaces/WF/WFArchiveFormat.ts]: Fixed up all the other bits but this. Will double check in the morning when I'm on my comp!

Comment (joshfarrant) [reply src/interfaces/WF/WFArchiveFormat.ts]: Great, thanks! 😎

Comment (gcordalis) [reply src/interfaces/WF/WFArchiveFormat.ts]: Fixed!

**Comment by gcordalis** [comment]
@joshfarrant thanks for the merge!

Sorry to pester, any chance we can get an NPM Publish soon?

**Comment by joshfarrant** [comment]
Yes of course. Meant to do it this morning before I left for work to be honest, but it slipped my mind. I'll do it shortly 👍

**Pull request merged**